pkg/cloud/amazon/s3_storage_test.go

```diff
@@ -276,13 +276,14 @@ func TestPutS3AssumeRole(t *testing.T) {
 		)
 	})
 
-	t.Run("role-chaining", func(t *testing.T) {
```

Comment:
Should we keep the old test here and in KMS to confirm that the old implementation of role chaining works as well?

Reply:
The way that I'm setting up the test requires the old chain without the external ID to fail. I thought I'd sort of test both by having one link in the chain with an external ID and another link without external ID.